src/main/webapp/src/app/admin-page/admin-page.component.ts

```diff
@@ -0,0 +1,341 @@
+  gardenPlantIdListError = '';
+  errorMessage = '';
+
+  /**
+   * Initializes the component based on provided arguments
+   *
+   * @param route Contains arguments.
+   */
+  constructor(private route: ActivatedRoute, private httpClient: HttpClient) {}
+
+  ngOnInit(): void {
+    const gardenIdArg = this.route.snapshot.paramMap.get('garden-id');
+    if (!gardenIdArg) {
+      this.gardenProfile = null;
+      this.errorMessage = 'No garden-id argument in the query string.';
+      return;
+    }
+    this.createAdminPage(gardenIdArg);
+  }
+
+  /**
+   * Gets garden information for the specified garden from
+   * the server. Returns an observable HTTP response.
+   *
+   * Performs GET: /garden/{garden}
+   *
+   * @param garden The garden reqested from the server.
+   * @return the http response.
+   */
+  getGardenInfo(garden: string): Observable<HttpResponse<Garden>> {
+    return this.httpClient.get<Garden>('/garden/' + garden, {
+      observe: 'response',
+      responseType: 'json',
+    });
+  }
+
+  /**
+   * Gets all users this garden contains.
+   *
+   * Performs GET: /garden/{garden}/user-list
+   *
+   * @return the http response.
+   */
+  getGardenUserList(garden: string): Observable<HttpResponse<Array<String>>> {
+    return this.httpClient.get<Array<String>>(
+      '/garden/' + garden + '/user-list',
+      {
+        observe: 'response',
+        responseType: 'json',
+      }
+    );
+  }
+
+  /**
+   * Gets all plants this garden contains.
+   *
+   * Performs GET: /garden/{garden}/plant-list
+   *
+   * @return the http response.
+   */
+  getGardenPlantList(garden: string): Observable<HttpResponse<Array<String>>> {
+    return this.httpClient.get<Array<String>>(
+      '/garden/' + garden + '/plant-list',
+      {
+        observe: 'response',
+        responseType: 'json',
+      }
+    );
+  }
+
+  /**
+   * Gets user information for the specified user from
+   * the server. Returns an observable HTTP response.
+   *
+   * Performs GET: /user/{user}
+   *
+   * @param user The user reqested from the server.
+   * @return the http response.
+   */
+  getUserInfo(user: string): Observable<HttpResponse<User>> {
+    return this.httpClient.get<User>('/user/' + user, {
+      observe: 'response',
+      responseType: 'json',
+    });
+  }
+
+  /**
+   * Gets plant information for the specified plant from
+   * the server. Returns an observable HTTP response.
+   *
+   * Performs GET: /plant/{plant}
+   *
+   * @param plant The plant reqested from the server.
+   * @return the http response.
+   */
+  getPlantInfo(plant: string): Observable<HttpResponse<Plant>> {
+    return this.httpClient.get<Plant>('/plant/' + plant, {
+      observe: 'response',
+      responseType: 'json',
+    });
+  }
+
+  /**
+   * Creates garden summary.
+   *
+   * @param garden The garden to create an admin page of.
+   */
+  createGardenSummary(garden: string): void {
+    this.getGardenInfo(garden).subscribe({
```

Comment:
This is also somewhere I'd imagine knowing more `rxjs` would help out.
This code creates a waterfall of first requesting garden info, and then user info.

Ideally we could request both in parallel, and [zip](https://www.learnrxjs.io/learn-rxjs/operators/combination/zip) over the results.

---

No need to fix this here, can always improve later if there is time.

Reply:
Didn't know about this, same basic response as above.